docs/docs/concepts/05-building-blocks/01-signals.md

```diff
@@ -74,24 +75,26 @@ The following state variables can be exchanged as signals:
 <!-- TODO: copy and link the markdown documentation of the state representation library directly in the programming reference -->
 
 AICA state signals are built on the
-open-source [`state_representation`](https://aica-technology.github.io/control-libraries/versions/v7.1.0/md__github_workspace_source_state_representation__r_e_a_d_m_e.html)
+open-source [`state_representation`](https://aica-technology.github.io/control-libraries/versions/9.0.1/md__2github_2workspace_2source_2state__representation_2_r_e_a_d_m_e.html)
 library for Cartesian and joint state classes in C++ and Python.
 
+<!-- TODO: all other state rep types too -->
+
 :::
 
 ## Custom messages
 
-The standard primitive and state message types are generally enough to cover the majority of messaging needs in
-an AICA application. Having a reduced message definition set is important to maximizing the modularity and compatibility
-of components. When components are connected by a signal in an application graph, the application interpreter will try
-to assert that the signals have a matching type.
+The set of basic and state signal types is generally enough to cover the majority of messaging needs in an AICA
+application. Having a reduced message definition set is important to maximizing the modularity and compatibility
+of components.
 
-However, any ROS 2 message can be implemented as a signal using the `custom` signal type. As long as the custom type
-between two connected components has the same name, the application will be valid.
+However, any ROS 2 message type can be implemented as a signal in an AICA application. The syntax for doing this is
```

Comment:
We say publishers/subscribers are signals, but here it feels like we now identify them with messages too. I see where we are going with this, but I think it's a bit inconsistent in its use.